### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Refactored code from the backend

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CarController.cpp [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
+void CarController::SetLightCmd(bool p_light) const
 {
-	
+	m_car->ctrl.lightCmd = p_light;
 }
 
 /// @brief			  Shows an element on the UI
 /// @param  p_element The element that should be shown
-void CarController::ShowUI(InterventionAction p_element)
+void CarController::ShowUI(InterventionAction p_element) const
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CarController.cpp:41]: ShowUI is kinda misleading since it can also sound the alarm

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h [hunk 2]
```diff
@@ -1,16 +1,19 @@
 #pragma once
 
+/// @brief The different tasks that can be performed
 typedef unsigned int Task;
 
 #define TASK_NO_TASK        0
 #define TASK_LANE_KEEPING   1
 #define TASK_SPEED_CONTROL  2
 
+/// @brief The different indicators that can be shown
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/ConfigEnums.h:10]: I'd reword it to something closer to "The different ways a user can be indicated about an intervention" since "shown" doesn't apply to auditory indicators

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl [hunk 3]
```diff
@@ -1,38 +1,54 @@
...
 /// @param  p_type The new type of interventions
-template<typename SocketBlackBox,typename SDAConfig>
+template<typename SocketBlackBox, typename SDAConfig>
 void TEMP_DECISIONMAKER::ChangeSettings(InterventionType p_type)
 {
     m_interventionExecutor = Config.SetInterventionType(p_type);
 }
 
 /// @brief         Changes the settings of how decisions should be made
 /// @param  p_type The new type of interventions
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl:50]: This comment does not apply to this method, param should be p_dataSetting

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by noutheesink — CHANGES_REQUESTED** [review]
Please change all tabs to spaces

**Review by noutheesink — APPROVED** [review]
All tabs should be changed to spaces but for now this is okay
